utils/oomparser/oomparser.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oomparser
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/cadvisor/utils"
+)
+
+var containerRegexp *regexp.Regexp = regexp.MustCompile(
+	`Task in (.*) killed as a result of limit of `)
+var lastLineRegexp *regexp.Regexp = regexp.MustCompile(
+	`(^[A-Z]{1}[a-z]{2} [0-9]{1,2} [0-9]{1,2}:[0-9]{2}:[0-9]{2}) .* Killed process ([0-9]+) \(([0-9A-Za-z_]+)\)`)
+var firstLineRegexp *regexp.Regexp = regexp.MustCompile(
+	`kernel: [[0-9]+.[0-9]+] [0-9A-Za-z_]+ invoked oom-killer:`)
+
+// struct to hold a list of OomInstance objects
+type OomParser struct {
+	oomInstances []*OomInstance
```

Comment:
lets remove the slice of all OOM instances and let the caller keep them if they want.